datus-snowflake/datus_snowflake/connector.py

```
@@ -0,0 +1,884 @@
+                        schemas.append(schema_name)
+
+                return schemas
+        except Exception as e:
+            logger.warning(f"Failed to get schemas using SHOW SCHEMAS, falling back to INFORMATION_SCHEMA: {e}")
+
+            # Fallback to INFORMATION_SCHEMA
+            select_table_name = (
+                "INFORMATION_SCHEMA.SCHEMATA" if not database_name else f'"{database_name}".INFORMATION_SCHEMA.SCHEMATA'
+            )
+
+            sql = f"SELECT SCHEMA_NAME FROM {select_table_name}"
```

Ruff (0.14.3)

Possible SQL injection vector through string-based query construction

(S608)